ckanext/data_qld/resource_freshness/validation.py

```diff
@@ -19,13 +22,12 @@ def validate_next_update_due(keys, flattened_data, errors, context):
         if next_update_due:
             next_update_due = tk.get_validator('isodate')(next_update_due, {})
             if next_update_due.date() <= dt.date.today():
-                raise tk.ValidationError({key: [tk._("Valid date in the future is required")]})
+                errors[keys].append(tk._("Valid date in the future is required"))
+        elif tk.get_endpoint()[1] == 'action':
+            flattened_data[keys] = h.recalculate_next_update_due_date(update_frequency)
         else:
-            # Recalculate only if the request is coming from the API action
-            if tk.get_endpoint()[1] == 'action':
-                flattened_data[keys] = h.recalculate_next_update_due_date(update_frequency)
-            else:
-                raise tk.ValidationError({key: [tk._("Missing value")]})
+            errors[keys].append(tk._('Missing value'))
+            raise StopOnError
```

Comment:
@tino097 Do we need to use StopOnError here? If we do, should we add do the other validation errors on this method?
Also StopOnError should be imported from toolkit